### ISSUE TITLE
Exception when no temperature reported  by disk

### ISSUE THREAD

**Issue by kstaniek** [colinodell/python-qnapstats#16]
Please provide the following information along with your issue report (replace anything inside of `<>`):

**Device Model Number**: TS-659

**QTS Version**: <4.2.6>

If you're reporting a `KeyError`, please also provide the XML response from your QNAP device.  You can easily obtain this by running the `debug.py` script.

**XML/Debug Output:**

~~~
DEBUG: Creating new session
DEBUG: POST to URL: http://172.31.20.200:8080/cgi-bin/authLogin.cgi
DEBUG: Request executed: 200
DEBUG: Response Text: <?xml version="1.0" encoding="UTF-8" ?>
<QDocRoot version="1.0">
<doQuick><![CDATA[]]></doQuick><is_booting><![CDATA[0]]></is_booting><mediaReady><![CDATA[1]]></mediaReady><SMBFW><![CDATA[0]]></SMBFW><authPassed><![CDATA[1]]></authPassed><authSid><![CDATA[ob24eucq]]></authSid><isAdmin><![CDATA[1]]></isAdmin><username><![CDATA[admin]]></username><groupname><![CDATA[administrators]]></groupname><ts><![CDATA[84872568]]></ts><fwNotice><![CDATA[0]]></fwNotice><SUID><![CDATA[caa72aa143fce9fd2ca58b98b9cacf58]]></SUID><title><![CDATA[]]></title><content><![CDATA[]]></content><psType><![CDATA[1]]></psType><showVersion><![CDATA[0]]></showVersion><show_link><![CDATA[1]]></show_link></QDocRoot>

DEBUG: GET from URL: http://172.31.20.200:8080/cgi-bin/disk/qsmart.cgi?func=all_hd_data
DEBUG: Appending access_token (SID: ob24eucq) to url
DEBUG: Request executed: 200
DEBUG: Response Text: <?xml version="1.0" encoding="UTF-8" ?>
<QDocRoot version="1.0">
<authPassed><![CDATA[1]]></authPassed><model><modelName><![CDATA[TS-659]]></modelName><internalModelName><![CDATA[TS-659]]></internalModelName><platform><![CDATA[TS-NASX86]]></platform><customModelName><![CDATA[]]></customModelName><displayModelName><![CDATA[TS-659]]></displayModelName><storage_v2>0</storage_v2>
</model>
<firmware><version><![CDATA[4.2.6]]></version><build><![CDATA[20170517]]></build><patch><![CDATA[0]]></patch><buildTime><![CDATA[2017/05/17]]></buildTime></firmware>
<specVersion><![CDATA[1.0]]></specVersion><hostname><![CDATA[SHAUN]]></hostname><DemoSiteSuppurt><![CDATA[no]]></DemoSiteSuppurt><customLogo><customFrontLogo><![CDATA[]]></customFrontLogo><customLoginLogo><![CDATA[]]></customLoginLogo></customLogo>
<HDTempWarnT><![CDATA[55]]></HDTempWarnT><HDTempErrT><![CDATA[60]]></HDTempErrT><Disk_Info><entry><Disk_Status><![CDATA[0]]></Disk_Status><HDNo><![CDATA[1]]></HDNo><Vendor><![CDATA[WDC     WD20EFRX-68EUZN082.0]]></Vendor><io_health><![CDATA[OK]]></io_health><Health><![CDATA[OK]]></Health><Capacity><![CDATA[1.82 TB]]></Capacity><Temperature><oC><![CDATA[38]]></oC><oF><![CDATA[100]]></oF></Temperature>
<LastTestSts><![CDATA[]]></LastTestSts><LastTestType><![CDATA[]]></LastTestType><LastTestTime><![CDATA[]]></LastTestTime><Model><![CDATA[WDC WD20EFRX-68EUZN0]]></Model><Serial><![CDATA[WD-WCC4M3AHY36Y]]></Serial><FirmVersion><![CDATA[82.00A82]]></FirmVersion><ATAVersion><![CDATA[ATA-9 ]]></ATAVersion><ATADescription><![CDATA[Exact ATA specification draft version not indicated]]></ATADescription></entry>
<entry><Disk_Status><![CDATA[0]]></Disk_Status><HDNo><![CDATA[2]]></HDNo><Vendor><![CDATA[WDC     WD20EFRX-68EUZN082.0]]></Vendor><io_health><![CDATA[OK]]></io_health><Health><![CDATA[OK]]></Health><Capacity><![CDATA[1.82 TB]]></Capacity><Temperature><oC><![CDATA[38]]></oC><oF><![CDATA[100]]></oF></Temperature>
<LastTestSts><![CDATA[]]></LastTestSts><LastTestType><![CDATA[]]></LastTestType><LastTestTime><![CDATA[]]></LastTestTime><Model><![CDATA[WDC WD20EFRX-68EUZN0]]></Model><Serial><![CDATA[WD-WCC4M5TXJDV9]]></Serial><FirmVersion><![CDATA[82.00A82]]></FirmVersion><ATAVersion><![CDATA[ATA-9 ]]></ATAVersion><ATADescription><![CDATA[Exact ATA specification draft version not indicated]]></ATADescription></entry>
<entry><Disk_Status><![CDATA[0]]></Disk_Status><HDNo><![CDATA[3]]></HDNo><Vendor><![CDATA[INTEL   SSDSA2M160G2GN  2CV1]]></Vendor><io_health><![CDATA[OK]]></io_health><Health><![CDATA[OK]]></Health><Capacity><![CDATA[149.05 GB]]></Capacity><Temperature><oC><![CDATA[]]></oC><oF><![CDATA[]]></oF></Temperature>
<LastTestSts><![CDATA[]]></LastTestSts><LastTestType><![CDATA[]]></LastTestType><LastTestTime><![CDATA[]]></LastTestTime><Model><![CDATA[INTEL SSDSA2M160G2GN]]></Model><Serial><![CDATA[CVPO13060046160AGN]]></Serial><FirmVersion><![CDATA[2CV102G9]]></FirmVersion><ATAVersion><![CDATA[ATA-7 (ATA133)]]></ATAVersion><ATADescription><![CDATA[ATA/ATAPI-7 T13 1532D revision 1]]></ATADescription></entry>
<entry><Disk_Status><![CDATA[-5]]></Disk_Status><HDNo><![CDATA[4]]></HDNo><Vendor><![CDATA[]]></Vendor><io_health><![CDATA[Abnormal]]></io_health><Health><![CDATA[Abnormal]]></Health><Capacity><![CDATA[]]></Capacity><Temperature><oC><![CDATA[]]></oC><oF><![CDATA[]]></oF></Temperature>
<LastTestSts><![CDATA[]]></LastTestSts><LastTestType><![CDATA[]]></LastTestType><LastTestTime><![CDATA[]]></LastTestTime><Model><![CDATA[]]></Model><Serial><![CDATA[]]></Serial><FirmVersion><![CDATA[]]></FirmVersion><ATAVersion><![CDATA[]]></ATAVersion><ATADescription><![CDATA[]]></ATADescription></entry>
<entry><Disk_Status><![CDATA[-5]]></Disk_Status><HDNo><![CDATA[5]]></HDNo><Vendor><![CDATA[]]></Vendor><io_health><![CDATA[Abnormal]]></io_health><Health><![CDATA[Abnormal]]></Health><Capacity><![CDATA[]]></Capacity><Temperature><oC><![CDATA[]]></oC><oF><![CDATA[]]></oF></Temperature>
<LastTestSts><![CDATA[]]></LastTestSts><LastTestType><![CDATA[]]></LastTestType><LastTestTime><![CDATA[]]></LastTestTime><Model><![CDATA[]]></Model><Serial><![CDATA[]]></Serial><FirmVersion><![CDATA[]]></FirmVersion><ATAVersion><![CDATA[]]></ATAVersion><ATADescription><![CDATA[]]></ATADescription></entry>
<entry><Disk_Status><![CDATA[0]]></Disk_Status><HDNo><![CDATA[6]]></HDNo><Vendor><![CDATA[WDC     WD30EFRX-68EUZN080.0]]></Vendor><io_health><![CDATA[OK]]></io_health><Health><![CDATA[OK]]></Health><Capacity><![CDATA[2.73 TB]]></Capacity><Temperature><oC><![CDATA[37]]></oC><oF><![CDATA[98]]></oF></Temperature>
<LastTestSts><![CDATA[]]></LastTestSts><LastTestType><![CDATA[]]></LastTestType><LastTestTime><![CDATA[]]></LastTestTime><Model><![CDATA[WDC WD30EFRX-68EUZN0]]></Model><Serial><![CDATA[WD-WMC4N2255621]]></Serial><FirmVersion><![CDATA[80.00A80]]></FirmVersion><ATAVersion><![CDATA[ATA-9 ]]></ATAVersion><ATADescription><![CDATA[Exact ATA specification draft version not indicated]]></ATADescription></entry>
</Disk_Info>
</QDocRoot>

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/klstanie/.virtualenvs/bmw/lib/python3.6/site-packages/qnapstats/qnap_stats.py", line 193, in get_smart_disk_health
    "temp_c": int(disk["Temperature"]["oC"]),
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
>>>
~~~

**Comment by colinodell** [comment]
Fixed in [0.2.6](https://github.com/colinodell/python-qnapstats/releases/tag/0.2.6) (which I'll be updating in HASS as well).

**Comment by colinodell** [comment]
Thank you so much for providing that debug information! :100:
